xdsl/dialects/affine.py

```diff
@@ -69,14 +70,13 @@ def from_region(
             upper_bound = IntegerAttr.from_index_int_value(upper_bound)
         if isinstance(step, int):
             step = IntegerAttr.from_index_int_value(step)
-        result_types = [SSAValue.get(op).typ for op in operands]
```

Comment:
It seems that this is not always the case, the MLIR For builder only has the iteration value types as the result types, not the lb/ub value types, but they are all operands

Reply:
are you referring to

```cpp
for (Value val : iterArgs)
  result.addTypes(val.getType());
```

from `AffineForOp::build`?

Reply:
yep, I'm not really sure that they're equivalent but it seems that they are? I've not actually used the CPP version before